runtime/parser/declaration.go

```diff
@@ -901,7 +896,13 @@ func parseCompositeOrInterfaceDeclaration(
 			p.next()
 			continue
 		} else {
-			identifier = p.tokenToIdentifier(p.current)
+			ctx := fmt.Sprintf("following %s declaration", compositeKind.Keyword())
+			nonReserved, err := p.nonReservedIdentifier(ctx)
```

Comment:
This always formats/allocates, even if the error message is not used. Maybe make this a closure that computes the message lazily.